chef/cookbooks/provisioner/templates/suse/crowbar_register.erb

```diff
@@ -46,7 +50,9 @@ while test $# -gt 0; do
         -h|--help|--usage|-\?) usage ;;
         -v|--verbose) CROWBAR_VERBOSE=1 ;;
         -f|--force) CROWBAR_FORCE=1 ;;
+        <% if !node[:provisioner][:keep_existing_hostname] -%>
         -k|--keep-existing-hostname) KEEP_EXISTING_HOSTNAME=1 ;;
+        <%end%>
```

Comment:
Can you add spaces around `end`? :-) Also, use `-%>` to close the tag (and avoid a new line).